scipy/signal/filter_design.py

```diff
@@ -248,6 +248,97 @@ def freqz(b, a=1, worN=None, whole=0, plot=None):
     return w, h
 
 
+def group_delay(b, a=1, w=None, whole=False):
+    r"""Compute the group delay of a digital filter.
+
+    The group delay measures by how many samples amplitude envelops of
+    various spectral components of a signal are delayed by a filter.
+    It is formally defined as::
+
+               d        jw
+     D(w) = - -- arg H(e)
+              dw
+
+    Where continuous (unwrapped) phase response is used for differentiation.
+
+    Parameters
+    ----------
+    b : array_like
+        Numerator coefficients of a filter transfer function.
+    a : array_like, optional
+        Denominator coefficients of a filter transfer function.
+    w : {None, int, array-like}, optional
+        If None (default), then compute at 512 frequencies equally spaced
+        around the unit circle.
+        If a single integer, then compute at that many frequencies.
+        If array, compute the delay at the frequencies given
+        (in radians/sample).
+    whole : bool, optional
+        Normally, frequencies are computed from 0 to the Nyquist frequency,
+        pi radians/sample (upper-half of unit-circle).  If `whole` is True,
+        compute frequencies from 0 to `2*pi` radians/sample.
+
+    Returns
+    -------
+    w : ndarray
+        The normalized frequencies at which the group delay was computed,
+        in radians/sample.
+    gd : ndarray
+        The group delay.
+
+    Notes
+    -----
+    The similar function in MATLAB is called `grpdelay`.
+
+    If the transfer function :math:`H(z)` has zeros or poles on the unit
+    circle, the group delay at corresponding frequencies is undefined.
+    When such case arises the warning is raised and the group delay
+    is set to 0.
+
+    References
+    ----------
+    Numerical Computation of Group Delay
+    https://ccrma.stanford.edu/~jos/fp/Numerical_Computation_Group_Delay.html
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> b, a = signal.iirdesign(0.1, 0.3, 5, 50, ftype='cheby1')
+    >>> w, gd = signal.group_delay(b, a)
+
+    >>> import matplotlib.pyplot as plt
+    >>> plt.title('Digital filter group delay')
+    >>> plt.plot(w, gd)
+    >>> plt.ylabel('Group delay [samples]')
+    >>> plt.xlabel('Frequency [rad/sample]')
+    >>> plt.show()
+
+    """
+    if w is None:
+        w = 512
```

Comment:
style nitpicks: 
- blank lines after the end of an ``if``-block
- there are a few more function calls to add ``np.`` to (``atleast_1d``, ``exp``, ``absolute``, ``real``.